apps/namadillo/src/atoms/integrations/atoms.ts

```diff
@@ -53,31 +54,38 @@ export const ibcTransferAtom = atomWithMutation(() => {
 });
 
 export const assetBalanceAtomFamily = atomFamily(
-  ({ chain, sourceAddress, assets }: AssetBalanceAtomParams) => {
+  ({ chain, walletAddress, assets }: AssetBalanceAtomParams) => {
     return atomWithQuery(() => ({
-      queryKey: ["assets", sourceAddress, chain?.chain_id, assets],
+      queryKey: ["assets", walletAddress, chain?.chain_id, assets],
       ...queryDependentFn(async () => {
         const assetsBalances = await queryAndStoreRpc(
           chain!,
           async (rpc: string) => {
-            return await queryAssetBalances(sourceAddress!, rpc);
+            return await queryAssetBalances(walletAddress!, rpc);
           }
         );
         return mapCoinsToAssets(assetsBalances, assets!);
-      }, [!!sourceAddress, !!chain]),
+      }, [!!walletAddress, !!chain]),
     }));
   },
   (prev, current) => {
     return Boolean(
       !current.chain ||
-        !current.sourceAddress ||
+        !current.walletAddress ||
         (prev.chain?.chain_id === current.chain?.chain_id &&
-          prev.sourceAddress === current.sourceAddress)
+          prev.walletAddress === current.walletAddress)
     );
   }
 );
 
-export const knownChainsAtom = atom((get) => {
-  const settings = get(settingsAtom);
-  return getKnownChains(settings.enableTestnets);
-});
+export const knownChainsAtom = atom<Record<string, ChainRegistryEntry>>(
+  (get) => {
+    const settings = get(settingsAtom);
+    const knownChains = getKnownChains(settings.enableTestnets);
+    const map: Record<string, ChainRegistryEntry> = {};
+    knownChains.forEach((chain) => {
+      map[chain.chain.chain_id] = chain;
+    });
+    return map;
```

Comment:
We are executing `Object.value(knownChains` four times that actaully reverts this map

![Screenshot 2024-10-22 at 17 10 08](https://github.com/user-attachments/assets/e2c04f4f-c7b9-4009-ab79-fad57ab43535)

I believe it'll be better if we return the original array and create a new atom for the map, that derivates from this atom

Or we can use a helper function like `_.keyBy` https://lodash.com/docs/4.17.15#keyBy directly in the component

Reply:
I've renamed a bit the atoms. Left the registry as the map structure with more info, and the availableChains as a list of chains